streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -422,11 +422,13 @@ private OptimizableRepartitionNode createRepartitionNode(final String repartitio
                                                              final Serde valueSerde) {
 
         final OptimizableRepartitionNode.OptimizableRepartitionNodeBuilder repartitionNodeBuilder = OptimizableRepartitionNode.optimizableRepartitionNodeBuilder();
+        final boolean userProvidedName = !repartitionTopicName.contains("KSTREAM-");
```

Comment:
Same thing here, at this point we have lost context if the user has provided the name as this is from the original logical plan.